### PR TITLE
Implement CRUD

### PR DESCRIPTION
**Pull request by mtraneis**
Added a new set of CRUD APIs for RequestPolicy at: /request-policy-storage/request-policies
- created new database table "request_policy" 
- added corresponding ramls and json schemas
- added request-type.json schema for request types, which can be reused elsewhere.
- the only business logic implemented is validation that request-policy name is required and unique
- implementation is based on previous CRUD implementations in the module
- all error messages are wrapped in the Error JSON structure. 

Cannot add tests to cover all the exception and negative cases because it's virtually impossible to trigger errors for them.

### CODE REVIEW

**Comment by mreno-EBSCO** [comment]
@mtraneis did you really mean to assign the pull request to us instead of setting us as reviewers? 😄

**Comment by mtraneis** [comment]
@mreno-EBSCO Yes, at 5am my eyes couldn't discern the difference between assignees and reviewers.  I made you reviewers now.

**Comment by mtraneis** [comment]
@mreno-EBSCO Addressed the issues you raised in the last commit.

**Comment by mtraneis** [comment]
@marcjohnson-kint Verified that metadata was included in the schema and a test for it is added.

File: ramls/request-policy-storage.raml [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+    delete:
+      is: [language]
+      responses:
+        204:
+          description: "All request policies deleted"
+        500:
+          description: "Internal server error, e.g. due to misconfiguration"
+          body:
+            application/json:
+              example: "Internal server error, contact administrator"
```
Comment (dcrossleyau) [inline ramls/request-policy-storage.raml]: @mtraneis This inline example not matching the content-type is what is causing "raml-cop" to notify the error on the PR.
